lib/dep-graph-builders/npm-lock-v2/index.ts

```diff
@@ -444,3 +443,42 @@ const checkOverrides = (
   }
   return;
 };
+
+// Here we have to match our pkg to
+// possible keys in the overrides object
+const matchOverrideKey = (
+  overrides: Overrides,
+  pkg: { name: string; version: string },
+): string | null => {
+  if (overrides[pkg.name]) {
+    return overrides[pkg.name];
+  }
+
+  const overrideKeysNameToVersions = Object.keys(overrides).reduce(
+    (acc, key) => {
+      const splitNameAndVersion = key.split('@');
+      const versionSpec = splitNameAndVersion.pop();
+      const name = splitNameAndVersion.join('@');
+      const versionsArr = acc[name]
+        ? [...acc[name], versionSpec]
+        : [versionSpec];
+      return { ...acc, [name]: versionsArr };
+    },
+    {},
+  );
+
+  if (overrideKeysNameToVersions[pkg.name]) {
+    for (const versionSpec of overrideKeysNameToVersions[pkg.name]) {
```

Comment:
why are we increasing computation time twice?

```
const keysNameToVersion = overrideKeysNameToVersions[pkg.name];

then reuse keysNameToVersion both for line 470 and line 471?
```

Reply:
Fair enough, was an extra hash lookup so didn't delve in too much